src/debug/ee/debugger.cpp

```diff
@@ -16732,6 +16732,18 @@ void *DebuggerHeap::Alloc(DWORD size)
     ret = pCanary->GetUserAddr();
 #endif
 
+#ifdef FEATURE_PAL
+    // We don't have executable heap in PAL, but we still need to allocate 
+    // executable memory, that's why have change protection level for 
+    // each allocation. 
+    // TODO: We need to look how JIT solves this problem.
+    DWORD unusedFlags;
+    if (!VirtualProtect(ret, size, PAGE_EXECUTE_READWRITE, &unusedFlags))
```

Comment:
Ah, right. But then we need to handle the cases when we need to set executable permissions for DebuggerHeap allocated memory differently. Leaking the executable permissions is creating a security hole.
According to @jkotas, the executable permission handling is only needed by the SharedPatchBypassBuffer class and by the DebuggerEval::m_breakpointInstruction. 
Could you please create a github issue to handle these two cases explicitly and to remove the executable permissions setting from the DebuggerHeap?